File: Grind169/Week1/704-Binary-Search/solution.py
from typing import List


class Solution:
    def search(self, nums: List[int], target: int) -> int:
        # Binary Search
        found = -1

        low = 0
        high = len(nums) - 1

        while low <= high:
            middle = low + (high - low) // 2

            if nums[middle] == target:
                return middle
            elif nums[middle] > target:
                high = middle - 1
            else:
                low = middle + 1

        return found

File: Grind169/Week1/704-Binary-Search/solution.cpp
#include <vector>

using namespace std;

class Solution
{
public:
    int search(vector<int> &nums, int target)
    {
        int low = 0;
        int high = nums.size() - 1;
        int middle;

        while (low <= high)
        {
            middle = low + (high - low) / 2;

            if (nums[middle] == target)
            {
                return middle;
            }
            else if (nums[middle] > target)
            {
                high = middle - 1;
            }
            else
            {
                low = middle + 1;
            }
        }
        return -1;
    }
};